src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs

```diff
@@ -268,5 +270,45 @@ private static string GetExpectedRedirectUri(string clientId)
         {
             return $"ms-appx-web://microsoft.aad.brokerplugin/{clientId}";
         }
+
+        /// <summary>
+        /// Converts to MSAL Account Id or Null
+        /// </summary>
+        /// <param name="wamAccount"></param>
+        /// <param name="clientID"></param>
+        public static IAccount ConvertToMsalAccount(
+                NativeInterop.Account wamAccount, 
+                string clientID)
+        {
+            IAccount runtimeAccount;
+
+            try
+            {
+                string homeAccountId = GetHomeAccountId(wamAccount);
+
+                runtimeAccount = new Account(
+                    wamAccount.AccountId,
+                    wamAccount.UserName,
+                    wamAccount.Environment,
+                    new Dictionary<string, string>() { { clientID, wamAccount.AccountId } });
+
+            }
+            catch (Exception ex)
+            {
+                throw new MsalServiceException("wam_failed", $"Could not convert into MSAL Account. {ex.Message}");
+            }
+
+            return runtimeAccount;
+        }
+
+        /// <summary>
+        /// Gets the home account id from client info
+        /// </summary>
+        /// <returns></returns>
+        public static string GetHomeAccountId(NativeInterop.Account account)
+        {
+            string homeAccId = ClientInfo.CreateFromJson(account.ClientInfo).ToAccountIdentifier();
```

Comment:
I think you can get the homeAccId easier, without the JSON stuff. I am undre the impression that Sam exposed some properties on the account object.